src/osd/OSD.cc

```diff
@@ -285,7 +285,7 @@ OSDService::~OSDService()
 {
   delete objecter;
 
-  for (auto f : objecter_finishers) {
+  for (auto& f : objecter_finishers) {
```

Comment:
i think a better way is to use `vector<unique_ptr<Finisher>>` instead of plain `vector<Finisher*>` for managing the life cycles of `Finisher` instances.

Reply:
I agree - but that's a larger fix. Whenever I'll have a chance to refactor that file as a whole.

Reply:
no problem i will do this.

Reply:
@ronen-fr could you please help review #32007?

Reply:
done. And - thanks